### PR TITLE
fixed NDPD-623 and added filtering staff and instructor users from xb…

### PR DESCRIPTION
**Pull request by kotky**
…lock grading table

### CODE REVIEW

File: edx_sg_block/sga.py [hunk 1]
```diff
@@ -528,7 +530,7 @@ def enter_grade(self, request, suffix=''):
 
         state = json.loads(module.state)
         try:
-            score = int(score)
+            score = float(score)
```
Comment (vkaracic) [inline edx_sg_block/sga.py:532]: Why this change?

File: edx_sg_block/sga.py [hunk 2]
```diff
@@ -679,3 +681,10 @@ def require(assertion):
     """
     if not assertion:
         raise PermissionDenied
+
+
+def check_user_access(user, course):
+    for level in ['instructor', 'staff']:
+        if has_access(user, level, course):
+            return True
+    return False
```
Comment (vkaracic) [inline edx_sg_block/sga.py]: Nit: `return any(has_access(user, level, course) for level in ['instructor', 'staff'])`

File: edx_sg_block/static/js/src/edx_sga.js [hunk 3]
```diff
@@ -130,7 +130,7 @@ function StaffGradedXBlock(runtime, element) {
                 var max_score = row.parents('#grade-info').data('max_score');
                 var score = Number(form.find('#grade-input').val());
                 event.preventDefault();
-                if (!score) {
+                if (!score && (score !== 0)) {
```
Comment (vkaracic) [inline edx_sg_block/static/js/src/edx_sga.js:133]: `if (typeof score !== 'number')` ?

**Review by vkaracic — DISMISSED** [review]

File: edx_sg_block/static/js/src/edx_sga.js (hunk 3 above)
Comment (kotky) [reply edx_sg_block/static/js/src/edx_sga.js:133]: the only bad case is 0, everything else is fetched in else if statements

File: edx_sg_block/sga.py (hunk 1 above)
Comment (kotky) [reply edx_sg_block/sga.py:532]: since the database field is float and gradebook is working with float values

File: edx_sg_block/sga.py (hunk 2 above)
Comment (kotky) [reply edx_sg_block/sga.py]: 👍

File: edx_sg_block/sga.py [hunk 4]
```diff
@@ -14,6 +14,7 @@
 from functools import partial
 
 from courseware.models import StudentModule
+from courseware.access import has_access
```
Comment (marjev) [inline edx_sg_block/sga.py]: nitpick: alphabetize

File: edx_sg_block/sga.py (hunk 2 above)
Comment (marjev) [reply edx_sg_block/sga.py]: DRY baby DRY, there should be a method semantically same as this one. You should search for `is_staff_or_instructor_on_course` and reuse that method.

**Review by marjev — CHANGES_REQUESTED** [review]

File: edx_sg_block/sga.py (hunk 4 above)
Comment (kotky) [reply edx_sg_block/sga.py]: sure

**Review by marjev — APPROVED** [review]

File: edx_sg_block/static/js/src/edx_sga.js (hunk 3 above)
Comment (vkaracic) [reply edx_sg_block/static/js/src/edx_sga.js:133]: Sure, but if you are outputting an error that says `Grade must be a number` it makes sense to check if the grade is a number.
